src/query/storage/index.go

```diff
@@ -136,26 +161,39 @@ func queryKey(m models.Matchers) []byte {
 	return key
 }
 
-// FetchQueryToM3Query converts an m3coordinator fetch query to an M3 query
-func FetchQueryToM3Query(fetchQuery *FetchQuery, cache *QueryConversionCache) (index.Query, error) {
+// FetchQueryToM3Query converts an m3coordinator fetch query to an M3 query.
+func FetchQueryToM3Query(
+	fetchQuery *FetchQuery,
+	cache *QueryConversionCache,
+) (index.Query, error) {
 	matchers := fetchQuery.TagMatchers
-	k := queryKey(matchers)
+	// If no matchers provided, explicitly set this to an AllQuery
+	if len(matchers) == 0 {
+		return index.Query{
+			// TODO: change this to an idx.AllQuery: https://github.com/m3db/m3/pull/1478
```

Comment:
is there something left server side before we can do this here?

Reply:
Nope, wrote this before AllQuery was pushed and forgot to update it :+1: